evm/src/main/java/org/hyperledger/besu/evm/operation/DataHashOperation.java

```diff
@@ -47,12 +47,12 @@ public DataHashOperation(final GasCalculator gasCalculator) {
 
   @Override
   public OperationResult execute(final MessageFrame frame, final EVM evm) {
-    int blobIndex = frame.popStackItem().toInt();
-    final Optional<List<Hash>> maybeHashes = frame.getVersionedHashes();
+    Bytes blobIndex = frame.popStackItem();
+    UInt256 blobIndexUInt256 = UInt256.fromBytes(blobIndex);
```

Comment:
I can optimize this later, but this will be slower than it should be.

(a) UInt256 is always slow.  Better to count leading zero bytes and convert to int if there are enough leading zero bytes.
(b) all actions other than the pop should be done after the check to see if versioned hashes are even present.

Reply:
good point, this will nearly always fit in something as small as even a byte.

Reply:
https://github.com/hyperledger/besu/blob/b3d3f54fff04e5aa2cccf96ab0a44ecfb580af31/evm/src/main/java/org/hyperledger/besu/evm/operation/ByteOperation.java#L39